client.go

```diff
@@ -477,13 +487,26 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 	if streamOptions.stderr == nil {
 		streamOptions.stderr = ioutil.Discard
 	}
-	cancelRequest := cancelable(c.HTTPClient, req)
+
+	// make a sub-context so that our active cancellation does not affect parent
+	ctx := streamOptions.context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	subCtx, cancelRequest := context.WithCancel(ctx)
+	defer cancelRequest()
```

Comment:
Like, there's a change in semantics here: we were only calling cancelRequest before on timeout, and now we're calling it every time we finish the function? Not saying you're wrong but I don't follow.